node/nodeBlocks.go

```diff
@@ -0,0 +1,58 @@
+package node
+
+import (
+	"encoding/hex"
+
+	apiBlock "github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/node/blockAPI"
+)
+
+// GetBlockByHash return the block for a given hash
+func (n *Node) GetBlockByHash(hash string, withTxs bool) (*apiBlock.APIBlock, error) {
+	decodedHash, err := hex.DecodeString(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlockProcessor := n.createAPIBlockProcessor()
```

Comment:
So we create these on the fly, for each request - but that is absolutely fine.

Reply:
yes. i think is not a problem